spec/section_spec.rb

```diff
@@ -44,24 +63,71 @@ class Page < SitePrism::Page; end
     expect(SitePrism::Section).to respond_to(:elements)
   end
 
-  it 'passes a given block to Capybara.within' do
-    expect(Capybara).to receive(:within).with('div')
+  describe 'instance' do
+    it 'passes a given block to Capybara.within' do
+      expect(Capybara).to receive(:within).with(locator)
 
-    SitePrism::Section.new(a_page, 'div') { 1 + 1 }
-  end
+      section_with_block
+    end
 
-  it 'does not require a block' do
-    expect(Capybara).not_to receive(:within)
+    it 'does not require a block' do
+      expect(Capybara).not_to receive(:within)
 
-    SitePrism::Section.new(a_page, 'div')
-  end
+      subject
```

Comment:
Named subjects would be nicer.  

I also prefer to assert afterwards.  You could add a spy on `within` in a before block and then move your asserts to the end for several of these tests.  Personal preference though.